packages/fetchai/connections/http_client/connection.py

```diff
@@ -0,0 +1,276 @@
+
+        :param agent_address: the address of the agent.
+        :param address: RESTful API hostname / IP address
+        :param port: RESTful API port number
+        :param excluded_protocols: this connection cannot handle messages adhering to any of the protocols in this set
+        :param restricted_to_protocols: this connection can only handle messages adhering to protocols in this set
+        """
+        self.agent_address = agent_address
+        self.address = address
+        self.port = port
+        self.restricted_to_protocols = restricted_to_protocols
+        self.in_queue = None  # type: Optional[asyncio.Queue]  # pragma: no cover
+        self.loop = None  # type: Optional[asyncio.AbstractEventLoop]  # pragma: no cover
+        self.excluded_protocols = excluded_protocols
+        self.is_stopped = True
+        logger.info("Initialised the HTTP client channel")
+
+    def connect(self):
+        """Connect."""
+        if self.is_stopped:
+            self.is_stopped = False
+            try:
+                logger.info(self.address)
+                response = requests.request(
+                    method="GET", url=f"http://{self.address}:{self.port}/status"
+                )
+                logger.debug("Status code is: {}".format(response.status_code))
+                assert response.status_code == 200, "Connection failed."
+            except Exception as e:  # pragma: no cover
+                logger.warning(str(e))
+                raise
+
+    def send(self, envelope: Envelope) -> None:
+        """
+        Send the envelope as a HTTP request, wait for its response, translate the response into an envelop, and sent it back to the agent.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        if self.excluded_protocols is not None:
+            if envelope.protocol_id in self.excluded_protocols:
+                logger.error(
+                    "This envelope cannot be sent with the oef connection: protocol_id={}".format(
+                        envelope.protocol_id
+                    )
+                )
+                raise ValueError("Cannot send message.")
+
+            if envelope is not None:
+                http_message = cast(
+                    HttpMessage, HttpSerializer().decode(envelope.message)
+                )
+                if http_message.performative == HttpMessage.Performative.REQUEST:
+                    response = requests.request(
+                        method=http_message.method,
+                        url=http_message.url,
+                        headers=http_message.headers,
+                        data=http_message.bodyy,
+                    )
+                    envelope = self.to_envelope(response)
+                    self.loop.call_soon_threadsafe(self.in_queue.put_nowait, envelope)
+                else:
+                    raise ValueError("The HTTPMessage performative must be a REQUEST.")
```

Comment:
We should not raise here, rather log a warning and drop the envelop. We don't want the agent to stop just because a wrong envelope is sent.

Reply:
Done